buildpack/build_test.go

```diff
@@ -27,1335 +27,1325 @@ import (
-								`default = true`+"\n",
-							filepath.Join(appDir, "launch-A-v1.toml"),
-						)
-						_, err := bpTOML.Build(buildpack.Plan{}, config, mockEnv)
-						h.AssertNotNil(t, err)
-						expected := "multiple default process types aren't allowed"
-						h.AssertStringContains(t, err.Error(), expected)
+						when("launch.toml", func() {
+							it("ignores process working directory and warns", func() {
+								h.Mkfile(t,
+									"[[processes]]\n"+
+										`working-dir = "/working-directory"`+"\n"+
+										`type = "some-type"`+"\n",
+									filepath.Join(appDir, "launch-A-v1.toml"),
+								)
+								br, err := descriptor.Build(buildpack.Plan{}, config, mockEnv)
+								h.AssertNil(t, err)
+								h.AssertEq(t, len(br.Processes), 1)
+								h.AssertEq(t, br.Processes[0].WorkingDirectory, "")
+								assertLogEntry(t, logHandler, "Warning: process working directory isn't supported in this buildpack api version. Ignoring working directory for process 'some-type'")
+							})
+						})
 					})
 				})
 			})
-
-			when("the launch, cache and build flags are in the top level", func() {
-				it("should error", func() {
-					mockEnv.EXPECT().WithPlatform(platformDir).Return(append(os.Environ(), "TEST_ENV=Av1"), nil)
-					h.Mkdir(t,
-						filepath.Join(layersDir, "A"),
-						filepath.Join(appDir, "layers-A-v1", "layer"),
-					)
-					h.Mkfile(t,
-						"build=true\ncache=true\nlaunch=true",
-						filepath.Join(appDir, "layers-A-v1", "layer.toml"),
-					)
-
-					_, err := bpTOML.Build(buildpack.Plan{}, config, mockEnv)
-					h.AssertNotNil(t, err)
-					expected := "the launch, cache and build flags should be in the types table"
-					h.AssertStringContains(t, err.Error(), expected)
-				})
-			})
-		})
-
-		when("buildpack api = 0.2", func() {
-			it.Before(func() {
-				bpTOML.API = "0.2"
-				mockEnv.EXPECT().WithPlatform(platformDir).Return(append(os.Environ(), "TEST_ENV=Av1"), nil)
-			})
-
-			it("should convert metadata version to top level version in the buildpack plan", func() {
-				bpPlan := buildpack.Plan{
-					Entries: []buildpack.Require{
-						{
-							Name:     "some-dep",
-							Metadata: map[string]interface{}{"version": "v1"},
-						},
-					},
-				}
-
-				_, err := bpTOML.Build(bpPlan, config, mockEnv)
-				if err != nil {
-					t.Fatalf("Unexpected error:\n%s\n", err)
-				}
-
-				testPlan(t,
-					[]buildpack.Require{
-						{
-							Name:     "some-dep",
-							Version:  "v1",
-							Metadata: map[string]interface{}{"version": "v1"},
-						},
-					},
-					filepath.Join(appDir, "build-plan-in-A-v1.toml"),
-				)
-			})
 		})
-
-		when("buildpack api < 0.5", func() {
-			it.Before(func() {
-				bpTOML.API = "0.4"
-			})
-
-			when("building succeeds", func() {
-				it.Before(func() {
-					mockEnv.EXPECT().WithPlatform(platformDir).Return(append(os.Environ(), "TEST_ENV=Av1"), nil)
-				})
-
-				it("should ensure the buildpack's layers dir exists and process build layers", func() {
```

Comment:
Moved to https://github.com/buildpacks/lifecycle/blob/b71e534defe5a2bca5eed1a738f7b8bc52522153/buildpack/build_test.go#L934-L961